addons/mail/static/src/js/mail_mention.js

```diff
@@ -0,0 +1,224 @@
+        this.partners = {};
+        this.keyup_list = [40, 38, 13, 35, 33, 34, 27];
+        this.partner_limit = 8;
+        this.typing_speed = 400;
+        this.min_charactor = 4;
+        this.search_string = '';
+        this.recent_string = '';
+    },
+
+    start: function() {
+        this._super.apply(this, arguments);
+        this.bind_events();
+    },
+    bind_events: function() {
+        var self = this;
+        this.$dropdown.insertAfter(this.$textarea);
+        
+        //textear event binding
+        this.$textarea.on('keyup', this.proxy('textarea_keyup'))
+        .on('click', function(e){ self.clear_dropdown(); })
+        .on('keydown', function(e){
+            var $active = self.$dropdown.find("li.active");
+            if(self.recent_string){
+                if (e.keyCode == 40) {
+                    $active.next().trigger("mouseover");
+                    return false;
+                } else if (e.keyCode == 38) {
+                    $active.prev().trigger("mouseover");
+                    return false;
+                } else if(e.keyCode == 13){
+                    self.find_n_replace(parseInt($active.find("span").attr("id")));
+                    return false;
+                }
+            }
+        });
+        
+        //dropdown event binding
+        this.$dropdown.on('mouseover', 'li', function(e) {
+            $(e.currentTarget).addClass('active').siblings().removeClass();
+        }).on('click', 'li', function(e){
+            self.find_n_replace(parseInt($(e.currentTarget).find("span").attr("id")));
+        });
+        this.clear_dropdown();
+    },
+    show_dropdown: function(){
+        var self = this;
+        var highlight = function(description){
+            if(description)
+            return description.replace(new RegExp(self.search_string, "gi"), function(str) {return _.str.sprintf("<b><u>%s</u></b>",str);});};
+
+        var res = _.map(self.search_keys[self.search_string], function(id) {
+                return {
+                    "id": id, 
+                    "name": highlight(self.partners[id]["name"]), 
+                    "email": highlight(self.partners[id]["email"])
+                    };
+        });
+        this.clear_dropdown();
+        this.$dropdown.find("ul").append(QWeb.render('Mention', {"result": res}));
+        this.$dropdown.find("li:first").addClass("active");
+        this.$dropdown.find("ul").show();
+    },
+
+    clear_dropdown: function() {
+        this.$dropdown.find("ul").hide();
+        this.$dropdown.find("ul").empty();
+    },
+    
+    search_n_store: function() {
+        var self = this;
+        if(!this.search_keys[this.search_string] && this.recent_string) {
+            this.model_res_partner.call("search_read", {
+                domain: ['|', ['name', 'ilike', self.search_string], ['email', 'ilike', self.search_string]],
+                fields: ['name', 'email'],
+                limit: this.partner_limit
```

Comment:
You should exclude from the result the partner already mentionned, like facebook is doing ;)